### PR TITLE
Exercise 2 - Hearthstone cards

### PR DESCRIPTION
**Pull request by marcosvonstremayr**
Module 2 of Hearthstone Cards. Read data from local JSON file, created a list of cards and implemented navigation between pages.

https://user-images.githubusercontent.com/109096201/180799689-444a24be-08d9-4850-8e72-fde8a8c2874a.mp4

### CODE REVIEW

File: lib/src/models/card_model.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  CardModel(this.cardId,
+      this.dbfId,
+      this.name,
+      this.cardSet,
+      this.type,
+      this.text,
+      this.health,
+      this.playerClass,
+      this.locale,
+      this.mechanics,);
```
Comment (GARAF004) [inline lib/src/models/card_model.dart]: I think you missed ctrl+alt+l to format code

File: lib/src/repositories/cards_repository.dart [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import 'dart:convert';
+import 'package:flutter/services.dart' show rootBundle;
+import 'package:hearthstone_cards/src/models/card_model.dart';
+
+class CardRepository {
+
+  Future<List<CardModel>> fetchCards() async {
+    List<CardModel> cardsModels = [];
+    var jsonText = await rootBundle.loadString('assets/cards_list.json');
```
Comment (GARAF004) [inline lib/src/repositories/cards_repository.dart]: This route could be added to a constant file

File: lib/src/ui/cards_list.dart [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+class CardList extends StatelessWidget {
+  const CardList({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return SafeArea(
+      child: Scaffold(
+        body: FutureBuilder<List<CardModel>>(
+            future: getCards(),
+          builder: (BuildContext context, AsyncSnapshot<List<CardModel>> snapshot) {
```
Comment (GARAF004) [inline lib/src/ui/cards_list.dart]: trailing comma

File: lib/src/repositories/cards_repository.dart [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import 'dart:convert';
+import 'package:flutter/services.dart' show rootBundle;
+import 'package:hearthstone_cards/src/models/card_model.dart';
```
Comment (Juliana-Dimeglio) [inline lib/src/repositories/cards_repository.dart]: This could be a relative import

File: lib/src/ui/cards_list.dart [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+              if (snapshot.hasError){
+                return const Text(Constants.errorMsg);
+              } else if (!snapshot.hasData) {
+                return const Center(
+                  child: CircularProgressIndicator(),
+                );
+              } else {
+                return ListView.builder(
+                  itemCount: snapshot.data!.length,
+                    itemBuilder: (BuildContext context, int index){
```
Comment (GARAF004) [inline lib/src/ui/cards_list.dart]: trailing comma

File: lib/src/ui/card_detail.dart [hunk 6]
```diff
@@ -0,0 +1,193 @@
+import 'package:flutter/material.dart';
+import 'package:hearthstone_cards/src/models/card_model.dart';
+import 'package:hearthstone_cards/utils/constants.dart';
```
Comment (Juliana-Dimeglio) [inline lib/src/ui/card_detail.dart]: relative import

File: lib/src/ui/single_card.dart [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+      child: Scaffold(
+        appBar: AppBar(
+          iconTheme: const IconThemeData(
+            color: Colors.black,
+          ),
+          title: const Text(
+              "Ysera Card",
+            style: TextStyle(
+              color: Colors.black,
+              fontSize: 40,
```
Comment (GARAF004) [inline lib/src/ui/single_card.dart]: Add fontSize to constants

File: lib/src/ui/cards_list.dart [hunk 8]
```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+import 'package:hearthstone_cards/src/models/card_model.dart';
+import 'package:hearthstone_cards/src/repositories/cards_repository.dart';
```
Comment (Juliana-Dimeglio) [inline lib/src/ui/cards_list.dart]: relative imports

File: lib/src/ui/single_card.dart (hunk 7 above)
Comment (Juliana-Dimeglio) [inline lib/src/ui/single_card.dart:33]: This could be a constant

**Review by Juliana-Dimeglio — CHANGES_REQUESTED** [review]

File: lib/src/ui/single_card.dart [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+        ),
+        floatingActionButton: FloatingActionButton(
+          onPressed: addLike,
+          backgroundColor: Constants.likeButtonColor,
+          child: const Icon(Icons.favorite),
+        ),
+      ),
+    );
+  }
+}
```
Comment (GARAF004) [inline lib/src/ui/single_card.dart]: Add new line at EOF

File: lib/utils/constants.dart [hunk 10]
```diff
@@ -27,4 +26,34 @@ abstract class Constants {
   static const String cardHealth = "Health: 12";
   static const String cardAbility =
       "Ability: At the end of your turn, add a Dream Card to your hand";
+  static const double iconPadding = 15;
+  static const double subtitleFontSize = 18;
+  static const double titleFontSize = 30;
+  static const String errorMsg = "An error has occurred";
+  static var tileColor = Colors.orange[50];
+  static var cardColor = Colors.orange[100];
```
Comment (GARAF004) [inline lib/utils/constants.dart]: Instead to do this, you can replace it whit Colors.orange.shade50 and Colors.orange.shade100

File: lib/src/models/card_model.dart (hunk 1 above)
Comment (marcosvonstremayr) [reply lib/src/models/card_model.dart]: Done

File: lib/src/repositories/cards_repository.dart (hunk 2 above)
Comment (marcosvonstremayr) [reply lib/src/repositories/cards_repository.dart]: Done

File: lib/src/ui/cards_list.dart (hunk 3 above)
Comment (marcosvonstremayr) [reply lib/src/ui/cards_list.dart]: Done

File: lib/src/ui/cards_list.dart (hunk 5 above)
Comment (marcosvonstremayr) [reply lib/src/ui/cards_list.dart]: Done

File: lib/src/ui/single_card.dart (hunk 7 above)
Comment (marcosvonstremayr) [reply lib/src/ui/single_card.dart]: Done

File: lib/src/repositories/cards_repository.dart (hunk 4 above)
Comment (marcosvonstremayr) [reply lib/src/repositories/cards_repository.dart]: Done

File: lib/src/ui/card_detail.dart (hunk 6 above)
Comment (marcosvonstremayr) [reply lib/src/ui/card_detail.dart]: Done

File: lib/src/ui/cards_list.dart (hunk 8 above)
Comment (marcosvonstremayr) [reply lib/src/ui/cards_list.dart]: Done

File: lib/src/ui/single_card.dart (hunk 9 above)
Comment (marcosvonstremayr) [reply lib/src/ui/single_card.dart]: Done

File: lib/utils/constants.dart (hunk 10 above)
Comment (marcosvonstremayr) [reply lib/utils/constants.dart]: Done

**Review by Juliana-Dimeglio — APPROVED** [review]

**Review by GARAF004 — APPROVED** [review]

**Review by LeonardoMolleker — APPROVED** [review]

**Pull request merged**